cli/register/servicewait.go

```diff
@@ -24,115 +105,193 @@ func WaitForService(org string, waitService string, waitTimeout int, pattern str
+		if len(servSpecArr) > 1 {
+			servSpecArr = servSpecArr[:len(servSpecArr)-1]
+
+			// exit if all services are started successfully
+			allSuccess, _ := ServiceAllSucess(servSpecArr)
+			if allSuccess {
+				DisplayServiceStatus(servSpecArr, false)
+				return
+			}
 		}
 	}
 
 	// If we got to this point, then there is a problem.
-	msgPrinter.Printf("Timeout waiting for service %v/%v to successfully start. Analyzing possible reasons for the timeout...", org, waitService)
+	msgPrinter.Printf("Timeout waiting for some services to successfully start. Analyzing possible reasons for the timeout...")
 	msgPrinter.Println()
 
-	// Let's see if we can provide the user with some help figuring out what's going on.
-	found := false
-	for _, serviceInstance := range services.Instances["active"] {
+	// Save services to an array based on their status
+	failedArr, failedDescArr, delayedArr, notDeployedArr := []serviceSpec{}, []string{}, []serviceSpec{}, []serviceSpec{}
 
-		// 1. Maybe the service is there but just hasnt started yet.
-		if serviceInstance.SpecRef == waitService && serviceInstance.Org == org {
-			msgPrinter.Printf("Service %v/%v is deployed to the node, but not executing yet.", org, waitService)
-			msgPrinter.Println()
-			found = true
+	// Let's see if we can provide the user with some help figuring out what's going on.
+	for i, ss := range servSpecArr {
+		if ss.status <= 0 {
+			found := false
+			for _, serviceInstance := range services.Instances["active"] {
+				// 1. Maybe the service is there but just hasnt started yet.
+				if serviceInstance.SpecRef == ss.name && serviceInstance.Org == ss.org {
+					found = true
 
-			// 2. Maybe the service has encountered an error.
-			if serviceInstance.ExecutionStartTime == 0 && serviceInstance.ExecutionFailureCode != 0 {
-				msgPrinter.Printf("Service %v/%v execution failed: %v.", org, waitService, serviceInstance.ExecutionFailureDesc)
-				msgPrinter.Println()
-				serviceFailed = true
-			} else {
-				msgPrinter.Printf("Service %v/%v might need more time to start executing, continuing analysis.", org, waitService)
-				msgPrinter.Println()
+					// 2. Maybe the service has encountered an error.
+					if serviceInstance.ExecutionStartTime == 0 && serviceInstance.ExecutionFailureCode != 0 {
+						failedArr = append(failedArr, ss)
+						failedDescArr = append(failedDescArr, serviceInstance.ExecutionFailureDesc)
+						servSpecArr[i].status = -1
+					} else {
+						delayedArr = append(delayedArr, ss)
+					}
+					break
+				}
 			}
-			break
 
+			// 3. The service might not even be there at all.
+			if !found {
+				notDeployedArr = append(notDeployedArr, ss)
+			}
 		}
 	}
 
-	// 3. The service might not even be there at all.
-	if !found {
-		msgPrinter.Printf("Service %v/%v is not deployed to the node, continuing analysis.", org, waitService)
+	// Print the status of each service
+	if len(failedArr) > 0 {
+		msgPrinter.Printf("The following services execution failed:")
```

Comment:
"The following services failed during execution:"

Reply:
Done.